server2/routes/confirmationRouter.js

```diff
@@ -0,0 +1,51 @@
+var express = require("express");
+var router = express.Router();
+var User = require("../models/user");
+var EmailToken = require("../models/emailToken");
+
+router
+    .route("/:token")
+    .get((req, res, next) => {
+        try {
+            const getToken = () => {
+                return new Promise((resolve, reject) => {
+                    resolve(EmailToken.findOne());
+                })
+            };
+
+            const verify = async () => {
+                const tokenObj = await getToken();
+                const token = tokenObj.token;
+
+                const jsonReceived = decodeURIComponent(req.params.token);
+                const unpacked = JSON.parse(jsonReceived);
+
+                const recToken = unpacked.token;
+
+                if (recToken == token) {        
+                    console.log("Token confirmed.");
+                    User.create(unpacked.user);
+                    
+                    // Confirm successful page
+                    res.send("Your Email is confirmed!");
+                } else {
+                    console.error("Token does not match!");
+                    // Confirm failed page
+                    res.send("Confirmation failed!\n" +
+                        "Some unexpected error has occured:\n\t"
+                        + err);
+                }
+                EmailToken.destroy(
+                    {where: {id: 1}}
```

Comment:
what does it mean? If multiple users register at the same time, the token from the first user who triggers the email confirmation will be erased if someone else send a wrong parameter on purpose?  This is totally unacceptable. The token itself should be able to trace back to the actual user or its identity (depends on how you implement the registration flow).